frontend/src/components/applyForm/types.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  EnumOptionsType,
+  FormContextType,
+  GenericObjectType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  UIOptionsType,
+} from "@rjsf/utils";
+
+import { HTMLAttributes } from "react";
+
+export interface FormData {
+  [key: string]: FormDataEntryValue;
+}
+
+export interface SetFormDataFunction {
+  (data: FormData): void;
+}
+
+export interface UiSchemaField {
+  type: "field";
+  definition: string;
+}
+
+export interface UiSchemaSection {
+  type: "section";
+  label: string;
+  name: string;
+  number: string;
+  children: (UiSchemaField | UiSchemaSection)[];
+}
+
+export type UiSchema =
+  | (UiSchemaField | UiSchemaSection)
+  | (UiSchemaField | UiSchemaSection)[];
+
+export type SchemaField = {
+  type: string;
+  title: string;
+  minLength?: number;
+  maxLength?: number;
+  format?: string;
+};
+
+export interface FormSchema {
+  $schema: string;
+  type: string;
+  properties: Record<string, SchemaField>;
+  required?: string[];
+}
+
+export type TextTypes =
+  | "text"
+  | "email"
+  | "number"
+  | "password"
+  | "search"
+  | "tel"
+  | "url";
+
+export interface UswdsWidgetProps<
```

Comment:
could be worth a comment in here linking to what you're basing this widget typing on

Reply:
Added, thanks.